src/App.tsx

```diff
@@ -1,14 +1,66 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import './app.scss';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState(0);
+  const [selectedOption, setSelectedOption] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const getTodosFromServer = async () => {
+    const todosFromServer = await getTodos();
+
+    setTodos(todosFromServer);
+  };
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const selectTodo = (todoId: number) => {
+    setSelectedTodoId(todoId);
+  };
+
+  const onCloseModal = () => {
+    setSelectedTodoId(0);
+  };
+
+  const selectedTodo = todos.find(todo => todo.id === selectedTodoId) as Todo;
+
+  const selectOption = (option: string) => {
+    setSelectedOption(option);
+  };
+
+  const handleQuery = (value: string) => {
+    setQuery(value);
+  };
+
+  let visibleTodos = [...todos];
+
+  if (selectedOption === 'completed') {
+    visibleTodos = visibleTodos.filter(todo => todo.completed);
+  }
+
+  if (selectedOption === 'active') {
+    visibleTodos = visibleTodos.filter(todo => !todo.completed);
+  }
```

Comment:
create eNum for that. You should do that when working with multiple string values